relay-profiling/src/sample/v2.rs

```diff
@@ -185,6 +190,52 @@ impl ProfileData {
         self.thread_metadata
             .retain(|thread_id, _| thread_ids.contains(thread_id));
     }
+
+    fn remove_idle_samples_at_the_edge(&mut self) {
```

Comment:
Would be good to add some docs, of why this is here and what it does for the future.